File: 0001/O(n).py
# _*_ encoding:utf-8 _*_

class Solution(object):
	def twoSum(self,nums,target):
		buff_dict={}
		for i in range(len(nums)):
			if nums[i] in buff_dict:
				return [buff_dict[nums[i]],i]
			else:
				buff_dict[target-nums[i]]=i
		return False

File: 0001/O(n).cpp
class Solution
{
  public:
    vector<int> twoSum(vector<int> &nums, int target)
    {
        map<int, int> buff_dict;
        vector<int> res;
        int len = nums.size();
        for (int i = 0; i < len; i++)
        {
            if (buff_dict.find(nums[i]) != buff_dict.end())
            {
                res.push_back(buff_dict[nums[i]]);
                res.push_back(i);
                return res;
            }
            else
            {
                buff_dict[target - nums[i]] = i;
            }
        }
        return res;
    }
};